app/components/primer/popover_component.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Primer
+  class PopoverComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    with_slot :message, :heading, :body, :button
+
+    class Message < ViewComponent::Slot
+      CARET_DEFAULT = :none
+      CARET_MAPPINGS = {
+        CARET_DEFAULT => "",
+        :bottom => "Popover-message--bottom",
+        :bottom_right => "Popover-message--bottom-right",
+        :bottom_left => "Popover-message--bottom-left",
+        :left => "Popover-message--left",
+        :left_bottom => "Popover-message--left-bottom",
+        :left_top => "Popover-message--left-top",
+        :right => "Popover-message--right",
+        :right_bottom => "Popover-message--right-bottom",
+        :right_top => "Popover-message--right-top",
+        :top_left => "Popover-message--top-left",
+        :top_right => "Popover-message--top-right"
+      }.freeze
+
+      def initialize(caret: CARET_DEFAULT, large: false, **kwargs)
+        @kwargs = kwargs
+        @kwargs[:classes] = class_names(
+          kwargs[:classes],
+          "Popover-message box-shadow-large",
```

Comment:
do we **always** want `box-shadow-large`? 🤔

Reply:
I wondered about this, I didn’t see some kind of box shadow mapping in classify. Guessing it hasn’t come up before. I can add a parameter to control this.

Reply:
I'm trying to add `box_shadow:` to classify.rb but I'm hitting trouble allowing for the default class when you say `box_shadow: true` and a different class when you say `box_shadow: :large`. I have these tests:

```ruby
    assert_generated_class("box-shadow",             { box_shadow: true })
    assert_generated_class("box-shadow-medium",      { box_shadow: :medium })
    assert_generated_class("box-shadow-large",       { box_shadow: :large })
    assert_generated_class("box-shadow-extra-large", { box_shadow: :extra_large })
    assert_generated_class("box-shadow-none",        { box_shadow: :none })
```

I can get `assert_generated_class("box-shadow",             { box_shadow: true })` to pass but then `assert_generated_class("box-shadow-medium",      { box_shadow: :medium })` fails, or vice versa. Any thoughts?

Reply:
Never mind, got it working in https://github.com/primer/view_components/pull/48/commits/4c2249f5ae1bcf1dab70269b0d9bbe55953e12fa!

Reply:
I noticed you added `box_shadow` to `BOOLEAN_MAPPINGS`
I'm pretty sure we don't allow a prop to be boolean and concat at the same time 🤔

Reply:
Yep, I removed that and handle it in #extract_hash now.